### ISSUE TITLE
Node status should include allocated resources info through the API

### ISSUE THREAD

**Issue by pwittrock** [kubernetes/kubernetes#27404]
See #25355.  Users want to be able to query the allocated resources for nodes and currently do this through `kubectl describe` which is both very inefficient as well as hard to parse the data out of.

**Comment by lavalamp** [comment]
@pwittrock please apply team/ux labels when filing issues against yourself! Thanks!

**Comment by pwittrock** [comment]
@lavalamp woops.  Thought I added the correct labels...

**Comment by AdoHe** [comment]
Since this is low priority, I would do it in the next milestone.

**Comment by dchen1107** [comment]
Wait, shouldn't pr https://github.com/kubernetes/kubernetes/pull/27365 address the issue here?

**Comment by AdoHe** [comment]
@dchen1107 I think pr #27365 addressed how to parse the data at kubectl side, but what @pwittrock means we should support this in API level.

**Comment by davidopp** [comment]
It seems like the pods that should be running on a machine (and their resources) are part of the "request" not the "status" ? If you're suggesting to put them in NodeStatus I'm not sure that is the right thing to do.

**Comment by jobevers** [comment]
/reopen

Unless there is a way to do this now, which I can't find.

**Comment by bungder** [comment]
It is reasonable to make it possible to retrieve allocated resources through API. Result of "Allocatable" minus "Capacity"  is neither amount of requests  nor amount of limits.

**Comment by SGrondin** [comment]
This is still a serious oversight. There doesn't seem to be a clean way to get this information programmatically other than parsing fragile "describe" output or iterating through pods on a node and querying their resource requests.

Can a contributor please reopen this?

**Comment by prasadjjoshi** [comment]
I created this script which parses 'kubectl describe nodes' & output the CPU/memory stats in a json since I needed it badly. This script can also be used with other automation code.
https://github.com/prasadjjoshi/k8-utils 
Feel free to use it.

**Comment by casidiablo** [comment]
I'm not even surprised anymore

**Comment by tnyeanderson** [comment]
Still needed, hate the autoclose bots...

**Comment by gengwg** [comment]
/reopen 
+1 need this!
